service/topology_coordinator.cc

```diff
@@ -1292,6 +1291,37 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         co_await update_topology_state(std::move(guard), std::move(updates), "Finished tablet migration");
     }
 
+    future<> handle_tablet_split_finalize(group0_guard g) {
+        std::vector<canonical_mutation> updates;
+        updates.reserve(_tables_to_finalize_split.size() * 2 + 1);
+
+        // Executes a global barrier to guarantee that any process (e.g. repair) holding stale version
+        // of token metadata will complete before we update topology.
+        auto guard = co_await global_tablet_token_metadata_barrier(std::move(g));
+
+        for (auto& table_id : _tables_to_finalize_split) {
```

Comment:
We won't need state in _tables_to_finalize_split if we invoke the plan maker here to get the most up to date plan. It's safer, since the plan will be based on the current tablet map. Table could be deleted, decisions can be revoked. Also, the less state the better.

Reply:
Makes sense, it's similar to how it's always better to invoke the compaction strategy right now rather than reusing a possibly outdated input sstable set. Things might change and you can get a fresher input.